src/session/Manager.cc

```diff
@@ -390,6 +159,17 @@ Connection* Manager::FindConnection(Val* v)
 	return conn;
 	}
 
+Connection* Manager::FindConnection(const zeek::detail::ConnKey& conn_key)
+	{
+	detail::Key key(&conn_key, sizeof(conn_key), false);
+
+	auto it = session_map.find(key);
+	if ( it != session_map.end() )
+		return static_cast<Connection*>(it->second);
```

Comment:
Hmm, can we use `dynamic_cast` here to double-check it's actually a `Connection`? Not sure about performance, but might be just fine. If it wasn't, we could at least still do that in debug compiles?

Reply:
This was the performance difference I DM'd you about. Changing this to a `dynamic_cast` is actually fairly costly.